processing/src/main/java/org/apache/druid/segment/UnnestCursorFactory.java

```diff
@@ -475,6 +453,73 @@ public static ColumnCapabilities computeOutputColumnCapabilities(
     }
   }
 
+  /**
+   * Converts a {@link CursorBuildSpec} to the base table {@link CursorBuildSpec}, ensuring that all required columns
+   * of are added to {@link CursorBuildSpec#getPhysicalColumns()} and the unnest column is added to {@link CursorBuildSpec#getVirtualColumns()}
+   */
+  @VisibleForTesting
+  public static CursorBuildSpec transformCursorBuildSpec(
```

Comment:
I don't see any usage outside of the `org.apache.druid.segment` package, should this drop the `public` visibility since it is marked as visiblefortesting?